### PR TITLE
fix: ignore the case of configured host

### LINKED ISSUE #3357: bug: failed to match the host which contains uppercase letters
here is a case:

```shell
$ curl -i http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "uri": "/hello",
    "hosts": ["A.COM"],
    "upstream": {
        "type": "roundrobin",
        "nodes": {
            "127.0.0.1:1980": 1
        }
    }
}'

$ curl http://127.0.0.1:9080/hello -H 'host: a.com'
{"error_msg":"404 Route Not Found"}

$ curl http://127.0.0.1:9080/hello -H 'host: A.COM'
{"error_msg":"404 Route Not Found"}
... ...

### PR DESCRIPTION
**Pull request by spacewander**
Fix #3357.
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: rockspec/apisix-master-0.rockspec [hunk 1]
```diff
@@ -43,7 +43,7 @@ dependencies = {
     "lua-resty-cookie = 0.1.0",
     "lua-resty-session = 2.24",
     "opentracing-openresty = 0.1",
-    "lua-resty-radixtree = 2.6",
+    "lua-resty-radixtree = 2.6.1",
```
Comment (Firstsawyou) [inline rockspec/apisix-master-0.rockspec:46]: `lua-resty-expr` also released v1.1.0, I think it can be updated by the way.

Comment (spacewander) [reply rockspec/apisix-master-0.rockspec:46]: Better to use separate PR to update lua-resty-expr, as the fix doesn't require to change it.

Comment (Firstsawyou) [reply rockspec/apisix-master-0.rockspec:46]: Yes.

**Review by moonming — APPROVED** [review]

File: apisix/router.lua [hunk 2]
```diff
@@ -32,6 +33,14 @@ local function filter(route)
         return
     end
 
+    if route.value.host then
+        route.value.host = str_lower(route.value.host)
```
[the remark this answers is not in the document]
Comment (tokers) [reply apisix/router.lua:37]: OK, forgot it, i found the answer: https://serverfault.com/questions/672984/is-the-hostname-part-of-https-urls-truly-case-insensitive

[the remark this answers is not in the document]
Comment (spacewander) [reply apisix/router.lua:37]: Yes, the host in HTTP should be case-insensitive.

**Review by tokers — APPROVED** [review]

**Pull request merged**